Add `--disable-network-filter` to skip workerd network filtering

frandiox opened the pull request:
When using Spin, we can't make queries to its private domain from workerd.
This is supposed to bypass the network filter made in workerd but for some reason we don't get `response.body` anymore with this change.

Still WIP.

jsaubry reviewed (approved):
I was able to test it, I just add to pass custom certs to node to make it work. I used `NODE_EXTRA_CA_CERTS=<path to .crt> npm run dev -- --disable-network-filter`

frandiox commented:
After testing a bit with @jsaubry , I've changed the approach and removed the original `--disable-network-filter` flag in favor of reusing Node's environment variables, since developers are already familiar with that.
